hoomd/mpcd/CollisionMethod.h

```diff
@@ -91,21 +100,74 @@ class PYBIND11_EXPORT CollisionMethod : public Autotuned
             }
         }
 
+    //! Get the rigid body definitions
+    std::shared_ptr<hoomd::md::ForceComposite> getRigid()
+        {
+        return m_rigid_bodies;
+        }
+
+    //! Set the rigid body definitions
+    void setRigid(std::shared_ptr<hoomd::md::ForceComposite> new_rigid)
+        {
+        m_rigid_bodies = new_rigid;
+        }
+
     protected:
     std::shared_ptr<SystemDefinition> m_sysdef;                //!< HOOMD system definition
     std::shared_ptr<hoomd::ParticleData> m_pdata;              //!< HOOMD particle data
     std::shared_ptr<mpcd::ParticleData> m_mpcd_pdata;          //!< MPCD particle data
     std::shared_ptr<const ExecutionConfiguration> m_exec_conf; //!< Execution configuration
 
-    std::shared_ptr<mpcd::CellList> m_cl;         //!< MPCD cell list
-    std::shared_ptr<ParticleGroup> m_embed_group; //!< Embedded particles
+    std::shared_ptr<mpcd::CellList> m_cl;                      //!< MPCD cell list
+    std::shared_ptr<ParticleGroup> m_embed_group;              //!< Embedded particles
+    std::shared_ptr<hoomd::md::ForceComposite> m_rigid_bodies; //!< definition for rigid bodies
 
     uint64_t m_period;        //!< Number of timesteps between collisions
     uint64_t m_next_timestep; //!< Timestep next collision should be performed
 
+    bool m_checked_collision_warnings; //!< True if collision related warnings have been checked
+
+    GPUArray<Scalar4> m_initial_velocity; //!< Initial velocities of the embedded particles
+    GPUArray<Scalar3> m_linmom_accum;     //!< Accumulated change in linear momentum of rigid bodies
+    GPUArray<Scalar3> m_angmom_accum; //!< Accumulated change in angular momentum of rigid bodies
+
     //! Check if a collision should occur and advance the timestep counter
     virtual bool shouldCollide(uint64_t timestep);
 
+    //! Check for issues related to applying collision to rigid bodies
+    void checkCollisionWarnings(uint64_t timestep);
+
+    //! Begin process of applying collisions to rigid bodies
+    void storeInitialEmbeddedGroupVelocities(uint64_t timestep);
+
+    //! Accumulate momenta changes of constituent particles of rigid bodies
+    void accumulateRigidBodyMomenta(uint64_t timestep);
+
+    //! Finish process of applying collisions to rigid bodies
+    void transferRigidBodyMomenta(uint64_t timestep);
+
+#ifdef ENABLE_MPI
+    GPUArray<Scalar3> m_linmom_accum_copybuf; //!< copy buffer for linear momentum
+    GPUArray<Scalar3> m_angmom_accum_copybuf; //!< copy buffer for angular momentum
+
+    /// The systems's communicator.
+    std::shared_ptr<Communicator> m_comm;
+#endif
```

Comment:
These related to the MPI implementation and so should probably be removed while we resolve that elsewhere.
```suggestion
```